scripts/build-windows.ps1

```diff
@@ -0,0 +1,175 @@
+    [Parameter()][String]$STATIC_LINK_SSL = "OFF",
+    [Parameter()][String]$CXX11 = "ON",
+    [Parameter()][String]$BUILD_APPS = "ON",
+    [Parameter()][String]$UNIT_TESTS = "OFF"
+)
+
+# make all errors trigger a script stop, rather than just carry on
+$ErrorActionPreference = "Stop"
+
+$projectRoot = (Join-Path $PSScriptRoot "/.." -Resolve)
+
+# if running within AppVeyor, use environment variables to set params instead of passed-in values
+if ( $Env:APPVEYOR ) { 
+    if ( $Env:PLATFORM -eq 'x86' ) { $DEVENV_PLATFORM = 'Win32' } else { $DEVENV_PLATFORM = 'x64' }
+    if ( $Env:APPVEYOR_BUILD_WORKER_IMAGE -eq 'Visual Studio 2019' ) { $VS_VERSION='2019' }
+    if ( $Env:APPVEYOR_BUILD_WORKER_IMAGE -eq 'Visual Studio 2015' ) { $VS_VERSION='2015' }
+    if ( $Env:APPVEYOR_BUILD_WORKER_IMAGE -eq 'Visual Studio 2013' ) { $VS_VERSION='2013' }
+
+    #if not statically linking OpenSSL, set flag to gather the specific openssl package from the build server into package
+    if ( $STATIC_LINK_SSL -eq 'OFF' ) { $Env:GATHER_SSL_INTO_PACKAGE = $true }
+
+    #if unit tests are on, set flag to actually execute ctest step
+    if ( $UNIT_TESTS -eq 'ON' ) { $Env:RUN_UNIT_TESTS = $true }
+    
+    $CONFIGURATION = $Env:CONFIGURATION
+
+    #appveyor has many openssl installations - place the latest one in the default location unless VS2013
+    if( $VS_VERSION -ne '2013' ){
+        Remove-Item -Path "C:\OpenSSL-Win32" -Recurse -Force -ErrorAction SilentlyContinue | Out-Null
+        Remove-Item -Path "C:\OpenSSL-Win64" -Recurse -Force -ErrorAction SilentlyContinue | Out-Null
+        Copy-Item -Path "C:\OpenSSL-v111-Win32" "C:\OpenSSL-Win32" -Recurse | Out-Null
+        Copy-Item -Path "C:\OpenSSL-v111-Win64" "C:\OpenSSL-Win64" -Recurse | Out-Null
+    }
+}
+
+# persist VS_VERSION so it can be used in an artifact name later
+$Env:VS_VERSION = $VS_VERSION
+
+# select the appropriate cmake generator string given the environment
+if ( $VS_VERSION -eq '2019' ) { $CMAKE_GENERATOR = 'Visual Studio 16 2019'; $MSBUILDVER = "16.0"; }
+if ( $VS_VERSION -eq '2015' -and $DEVENV_PLATFORM -eq 'Win32' ) { $CMAKE_GENERATOR = 'Visual Studio 14 2015'; $MSBUILDVER = "14.0"; }
+if ( $VS_VERSION -eq '2015' -and $DEVENV_PLATFORM -eq 'x64' ) { $CMAKE_GENERATOR = 'Visual Studio 14 2015 Win64'; $MSBUILDVER = "14.0"; }
+if ( $VS_VERSION -eq '2013' -and $DEVENV_PLATFORM -eq 'Win32' ) { $CMAKE_GENERATOR = 'Visual Studio 12 2013'; $MSBUILDVER = "12.0"; }
+if ( $VS_VERSION -eq '2013' -and $DEVENV_PLATFORM -eq 'x64' ) { $CMAKE_GENERATOR = 'Visual Studio 12 2013 Win64'; $MSBUILDVER = "12.0"; }
+
+# clear any previous build and create & enter the build directory
+$buildDir = Join-Path "$projectRoot" "_build"
+Write-Output "Creating (or cleaning if already existing) the folder $buildDir for project files and outputs"
+Remove-Item -Path $buildDir -Recurse -Force -ErrorAction SilentlyContinue | Out-Null
+New-Item -ItemType Directory -Path $buildDir -ErrorAction SilentlyContinue | Out-Null
```

Comment:
Shouldn't failure from this action prevent rest of the script from execution? If this is going to create a new clean directory for the build, then failure to do so would result in problems later, wouldn't it.

Reply:
PowerShell fails things if you try to remove them and they do not already exist - so I eat the error because on first run the folder never exists (and it seems a bit overkill to test the existence of a folder just to delete it). 

There is an argument that the 'New-Item' should throw the error; but if it failed to make it the very next line ('Push-Directory') will fail because it does not exist and then it will abort. 

I set the '$ErrorActionPreference = "Stop"' flag at the top of the script that makes PowerShell more sensitive to failure; which is why there are some of the express 'allow fail' flags set for some things that might have non-fatal reasons to fail (e.g. if a console has the _build folder open, it will not delete totally or create but it should clean and it will allow you to then enter it).

Reply:
Ok. I overlooked it a bit.